### PR TITLE
[SPARK-23144][SS] Added console sink for continuous processing

### PR DESCRIPTION
**Pull request by tdas**
## What changes were proposed in this pull request?
Refactored ConsoleWriter into ConsoleMicrobatchWriter and ConsoleContinuousWriter.

## How was this patch tested?
new unit test

### CODE REVIEW

**Comment by tdas** [comment]
@jose-torres PTAL

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/sources/ConsoleWriter.scala [hunk 1]
```diff
@@ -20,45 +20,85 @@ package org.apache.spark.sql.execution.streaming.sources
...
+ * [[org.apache.spark.sql.execution.streaming.ConsoleSinkProvider]].
+ *
+ * This sink should not be used for production, as it requires sending all rows to the driver
+ * and does not support recovery.
+ */
+class ConsoleMicroBatchWriter(batchId: Long, schema: StructType, val options: DataSourceV2Options)
+  extends DataSourceV2Writer with ConsoleWriter {
+
+  override def commit(messages: Array[WriterCommitMessage]): Unit = {
+    printRows(messages, schema, s"Batch: $batchId")
```
Comment (jose-torres) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/sources/ConsoleWriter.scala]: Not part of this PR - but can you envision scenarios where microbatch.commit(messages) would be implemented differently than continuous.commit(batchId, messages)? Seeing this in context made me realize there's an obvious strategy to unify the streaming writer APIs.

**Comment by jose-torres** [comment]
LGTM
